### PR TITLE
Plugins Catalog: Install and show the latest compatible version of a plugin

### LINKED ISSUE #37934: Plugins Catalog: Add support for finding the compatible plugin version with the current running version of Grafana.
We should introduce an API that provides the most recent compatible version(s) based on the running Grafana version.

Tip: This code has existed in `grafana-cli` for some time and was recently moved to the backend. See [here](https://github.com/grafana/grafana/blob/2a4c1b1aa67260de69581ce5d85f607dadadaa21/pkg/plugins/manager/installer/installer.go#L435) for more details. We will likely just need to extract this code into something more reusable

### LINKED ISSUE #40803: PluginsCatalog: Incorrectly shows the latest version number and readme available rather than the installed version
**What happened**:
Browsed plugins using the plugins catalog and opened the grafana-image-renderer details page. I had version v3.0.1 installed of this plugin. Noticed that only information about the latest version (v3.2.1) number and readme was being displayed. 

image

**What you expected to happen**:
Information about currently installed version (v3.0.1) number and readme.

**How to reproduce it (as minimally and precisely as possible)**:
Install a plugin with a version not being the latest. Browse that plugin using the plugins catalog.

**Anything else we need to know?**:
If you're listing plugins you'll get a notice that there's an update available:
image

However, this information is not available when browsing the plugin details page. If we fix so that current version number and readme is rendered there must be a way to be able to read information about the latest version (the update) such as readme and potentially changelog. Maybe parts of this is separate issues/new features?

**Environment**:
- Grafana version: main

### LINKED ISSUE #41157: Plugins Catalog: align plugin version installation between backend and frontend.
## Description
While looking into the following issue: https://github.com/grafana/grafana/issues/37934 we realized that the logic for installing latest version is different on the backend and frontend in the plugins catalog.

The logic on the backend is the following:

- If the frontend do not pass a version when installing a plugin, backend will figure out the latest compatible version of the plugin for the current running grafana instance.
- If the frontend pass a version when installing a plugin, backend will "force" install that version and skip all compatibility checks.

The logic on the frontend:

We fetch the latest version of all plugins from grafana-com. When installing or updating a plugin we passes the latest available version to the backend. 

We should align this by never sending a version to backend when installing/updating a plugin.

## Todo

- [x] Never send the plugin version to the Grafana when installing a plugin.
- [x] Extend the data from Grafana backend about available updates by including the targeted version.

### LINKED ISSUE #41173: kubernetes plugins report not support for latest  version
**What happened**:
I installed kubeneter plugins in my local server, and my grafana version is 8.2.2, but I still get this warning this plugin doesn't support your version of Grafana.

image

image

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version:
- 
image

- Data source type & version:
- OS Grafana is installed on:
- installed  on ubuntu
- User OS & Browser:
- Grafana plugins:
- kuberneters-grafana-app
- Others:

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
PluginDetails page prefers the version from gcom (which is always the latest) over an installed plugin version. This is rather confusing for users as they cannot easily see which version they have installed. This PR addresses this by:

- [x] if the plugin is installed use it's version else fallback to the gcom version
- [x] make it clearer in version history tab which is the currently installed version 
- [x] show the latest compatible versions version number in the header if the plugin is not installed
- [x] use the `isCompatible` flag from the GCOM API to decide if a certain plugin version is compatible or not
- [x] add & update tests 

| Before | After |
| --- | --- |
| Screenshot 2021-10-27 at 17 26 35 | Screenshot 2021-10-27 at 17 26 02 |

**Which issue(s) this PR fixes**:

Fixes #40803
Fixes #41157
Fixes #37934
Fixes #41173

**Special notes for your reviewer**:

### CODE REVIEW

**Review by marefr — CHANGES_REQUESTED** [review]
Works as expected on the plugin details page/versions tab. Great work :+1: 

However, on the plugin listings page it now incorrectly shows an update of the currently installed version (I have 4.1.5 installed):
image

**Review by marefr — COMMENTED** [review]
Dependency information seems to always come from latest version at gcom?
image

No version to upgrade to on hover (or did you decide to remove that?):
image

Other than this seems to work as expected. Great work :+1: There's one more thing, but same thing on main branch so I'll bring that up outside of this.

File: public/app/features/plugins/admin/components/PluginDetailsHeader.tsx [hunk 1]
```diff
@@ -69,8 +69,8 @@ export function PluginDetailsHeader({ plugin, currentUrl, parentUrl }: Props): R
             </span>
           )}
 
-          {/* Latest version */}
-          {plugin.version && <span>{plugin.version}</span>}
+          {/* Version */}
+          {plugin.installedVersion ? <span>{plugin.installedVersion}</span> : <span>{plugin.latestVersion}</span>}
```
[the remark this answers is not in the document]
Comment (leventebalogh) [reply public/app/features/plugins/admin/components/PluginDetailsHeader.tsx]: Fixed in [b15a74d](https://github.com/grafana/grafana/pull/41003/commits/b15a74ddd1cd943da6c965677fd28b73a804da76) 👍

**Review by marefr — APPROVED** [review]
LGTM. Quite a lot of moving pieces/changes and references multiple issues. Maybe a bit risky to put out all of them in a patch release. Something to keep in mind moving forward, trying to solve one problem at a time could reduce risk and make it easier to review/test and allow certain pieces to be released in a patch and some in the next major/minor version. This is my personal opinions.

**Comment by marefr** [comment]
I would really like some others taking this for a test as well, since I'm not sure I've tested all possible combinations

**Comment by leventebalogh** [comment]
> Something to keep in mind moving forward, trying to solve one problem at a time could reduce risk and make it easier to review/test and allow certain pieces to be released in a patch and some in the next major/minor version. This is my personal opinions.

I absolutely agree mate, this was not just hard to review (because touching so many things) but also quite hard to test out as there are so many edge cases. I think I have managed to cover all of them, but for sure it would be cool if someone else could test it as well.

**Pull request merged**